src/sentry/migrations/0187_add_dashboardwidgetmetricsquery.py

```diff
@@ -0,0 +1,74 @@
+# Generated by Django 1.11.29 on 2021-04-22 10:24
+
+import django.db.models.deletion
+import django.utils.timezone
+from django.db import migrations, models
+
+import sentry.db.models.fields.array
+import sentry.db.models.fields.bounded
+import sentry.db.models.fields.foreignkey
+
+
+class Migration(migrations.Migration):
+    # This flag is used to mark that a migration shouldn't be automatically run in
+    # production. We set this to True for operations that we think are risky and want
+    # someone from ops to run manually and monitor.
+    # General advice is that if in doubt, mark your migration as `is_dangerous`.
+    # Some things you should always mark as dangerous:
+    # - Large data migrations. Typically we want these to be run manually by ops so that
+    #   they can be monitored. Since data migrations will now hold a transaction open
+    #   this is even more important.
+    # - Adding columns to highly active tables, even ones that are NULL.
+    is_dangerous = False
+
+    # This flag is used to decide whether to run this migration in a transaction or not.
+    # By default we prefer to run in a transaction, but for migrations where you want
+    # to `CREATE INDEX CONCURRENTLY` this needs to be set to False. Typically you'll
+    # want to create an index concurrently when adding one to an existing table.
+    # You'll also usually want to set this to `False` if you're writing a data
+    # migration, since we don't want the entire migration to run in one long-running
+    # transaction.
+    atomic = True
+
+    dependencies = [
+        ("sentry", "0186_add_externalactor"),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name="DashboardWidgetMetricsQuery",
+            fields=[
+                (
+                    "id",
+                    sentry.db.models.fields.bounded.BoundedBigAutoField(
+                        primary_key=True, serialize=False
+                    ),
+                ),
+                ("name", models.CharField(max_length=255)),
+                ("fields", sentry.db.models.fields.array.ArrayField(null=True)),
+                ("conditions", models.TextField()),
+                ("order", sentry.db.models.fields.bounded.BoundedPositiveIntegerField()),
+                ("date_added", models.DateTimeField(default=django.utils.timezone.now)),
+                ("groupby", sentry.db.models.fields.array.ArrayField(null=True)),
+                (
+                    "project",
+                    sentry.db.models.fields.foreignkey.FlexibleForeignKey(
+                        on_delete=django.db.models.deletion.CASCADE, to="sentry.Project"
```

Comment:
Foreign keys to busy tables can cause problems because of the difficulties aquiring a lock, we could add `db_constraint=False` on this to avoid the problem.
https://develop.sentry.dev/database-migrations/#foreign-keys